### PR TITLE
Mark TimeSeriesScalar as deprecated in all SDKs and documentation

### LINKED ISSUE #2367: Implement `deprecation_reason` attribute
Flatbuffers' builtin `deprecated` attribute is nice, but it's just a boolean.

We'd like to have an additional optional `deprecation_reason` attribute where we can explain why if we want to.

(Or maybe we just modify `reflection.fbs` to use `string` instead...? :grimacing:)

### LINKED ISSUE #4830: Add support for deprecation notices on Archetypes
As we begin to refactor and change some of our Archetypes it will be helpful to go through a 1-release deprecation cycle on things that would otherwise break user API usage.

When an existing archetype is being deprecated we should be able to apply a deprecated attribute to the archetype definition and code-gen deprecation notices for the Archetype constructors across all 3 languages.

In the future we should also be able to deprecate usage of individual component-fields within an archetype, but this will be a bit more involved and not needed immediately.

- Motivated by: https://github.com/rerun-io/rerun/issues/4816

The deprecated attribute should be a string with a message that can explain to users what to do, such as: "This archetype has been replaced by the new archetypes: Scalar and SeriesStyle"

### Python

In python we should follow the pattern established in: https://github.com/rerun-io/rerun/pull/3564/files
 - Add a `@deprecated` decorator to the class
 - Add information about the deprecation to the generated docstrings.
 
### C++
TODO

### Rust
TODO

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Our codegen can now deprecate archetypes
* Fixes #4830
* Does *not* address #2366
   * builtin attribute is for fields. Unclear if it can have text? 
   * Fixes #2367

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5102/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5102/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5102/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5102)
- [Docs preview](https://rerun.io/preview/b773eee58d95201fc1f16a023183294d739cf2f9/docs) 
- [Examples preview](https://rerun.io/preview/b773eee58d95201fc1f16a023183294d739cf2f9/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_types/src/archetypes/mod.rs [hunk 1]
```diff
@@ -65,6 +65,7 @@ pub use self::series_point::SeriesPoint;
 pub use self::tensor::Tensor;
 pub use self::text_document::TextDocument;
 pub use self::text_log::TextLog;
+#[allow(deprecated)]
 pub use self::time_series_scalar::TimeSeriesScalar;
```
Comment (teh-cmc) [inline crates/re_types/src/archetypes/mod.rs:69]: maybe move it down to the bottom then, separate from its up-to-date peers

**Review by teh-cmc — APPROVED** [review]

**Comment by Wumpf** [comment]
just noticed a few more places in the docs that should show `(deprecated)`

**Comment by teh-cmc** [comment]
- Marked it as closing https://github.com/rerun-io/rerun/issues/2366 anyway.

File: crates/re_types/definitions/rerun/attributes.fbs [hunk 2]
```diff
@@ -29,3 +29,13 @@ attribute "attr.rerun.override_type";
 ///
 /// This is used for example to scope blueprint types.
 attribute "attr.rerun.scope";
+
+/// Marks something as deprecated followed by a (non-optional!) migration note.
```
Comment (emilk) [inline crates/re_types/definitions/rerun/attributes.fbs:33]:
```suggestion
/// Marks something as deprecated followed by a (mandatory!) migration note.
```

:)

File: tests/python/chroma_downsample_image/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 numpy
-opencv-python
+opencv-python==
```
Comment (emilk) [inline tests/python/chroma_downsample_image/requirements.txt:2]: What does == without a rhs? O.o

**Review by emilk — APPROVED** [review]
Awesome to have this in!

File: tests/python/chroma_downsample_image/requirements.txt (hunk 3 above)
Comment (Wumpf) [reply tests/python/chroma_downsample_image/requirements.txt:2]: Oh. Oops!! Nothing to see here. (Long story but this change is wrong and shouldn't be here)

File: pixi.toml [hunk 4]
```diff
@@ -101,7 +101,7 @@ cpp-prepare-msvc = "cmake -G 'Visual Studio 17 2022' -B build-msvc -S ."
 [dependencies]
 attrs = ">=23.1.0"
 blackdoc = "0.3.8"
-clang-tools = ">=15,<16"     # clang-format
+clang-tools = "16.0.6"       # clang-format
```
Comment (jleibs) [inline pixi.toml:104]: Looks like this is maybe tickling a new lint error: https://github.com/rerun-io/rerun/actions/runs/7821224421/job/21337710444?pr=5102

Comment (Wumpf) [reply pixi.toml:104]: urgh. yeah I should undo all that stuff. the real fix was putting the codegen run on pixi. then again would be nice to run against newer clang, but maybe not in here!

Comment (Wumpf) [reply pixi.toml:104]: ... let's first try the forward-flight!

**Pull request merged**